go/tasks/plugins/array/awsbatch/job_definition.go

```diff
@@ -50,7 +50,12 @@ func EnsureJobDefinition(ctx context.Context, tCtx pluginCore.TaskExecutionConte
 		return nil, errors.Errorf(pluginErrors.BadTaskSpecification, "Tasktemplate does not contain a container image.")
 	}
 
-	role := awsUtils.GetRole(ctx, cfg.RoleAnnotationKey, tCtx.TaskExecutionMetadata().GetAnnotations())
+	role := awsUtils.GetRole(ctx, cfg.RoleSecurityContextKey, tCtx.TaskExecutionMetadata().GetSecurityContext())
```

Comment:
should we make this into a protobuf enum - `cfg.RoleSecurityContextKey`

Reply:
I am not inclined for this one. But for serviceAccountName may be? What do you think ?

Reply:
Why not for Role - AWS Batch, Sagemaker, Athena etc may all need a role, then how do we make it consistent

Reply:
done